### PR TITLE
Clarify client and issuer

### LINKED ISSUE #102: Add client in terminology
In section 5.2.3.3 `client` appears out of the blue. It is not clear what a `client` is.

### PR DESCRIPTION
**Pull request by SteveLasker**
Fixes #102, #186 and #189

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -192,6 +192,11 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
+
+: a process, user or another service that externally connects to the Transparency Service.
+A Client and Issuer are not synonymous as a Client may register Statements signed by a different Issuer identity.
```
Comment (aj-stein-nist) [inline draft-ietf-scitt-architecture.md:198]: Is a Verifier a Client? If not, does a Verifier use a Client? I bring this up because in my prior work on this I presented a updated diagram to confirm my understanding and I was corrected on this point. However, this seems an important place to clarify this and/or in the definition of Verifier, otherwise the clarification is not in the current draft of the document.

Comment (fournet) [reply draft-ietf-scitt-architecture.md:198]: I would simply write that the Issuer may or may not be the client that submits the signed claim to a TS for registration. (The proposed text suggests the client may also sign, which seems irrelevant here.)

Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:198]: If you read the original report in #102, your recommendation (from my perspective, others can feel free to disagree) further confuses the purpose of the original request for clarification: what a client (what is doing it) is unclear. Casually blurring the line between what is doing and who is doing (issuer, verifier has been unclear to me if it what or who) is part of the problem here.

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -192,6 +192,11 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
+
+: a process, user or another service that externally connects to the Transparency Service.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:197]: lets take a normative definition of client from another spec if we can... perhaps

https://datatracker.ietf.org/doc/html/rfc6749#section-1.1

```
client
      An application making protected resource requests on behalf of the
      resource owner and with its authorization.  The term "client" does
      not imply any particular implementation characteristics (e.g.,
      whether the application executes on a server, a desktop, or other
      devices).

```

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -544,12 +549,12 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
...
 
-The same Signed Statement may be independently registered in multiple Transparency Services.
 To register a Signed Statement, the Transparency Service performs the following steps:
 
-1. **Client authentication:** This is implementation-specific and MAY be unrelated to the Issuer identity.
-Signed Statements may be registered by a different party than their Issuer.
-1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity.
+1. **Client authentication:** A Client authenticates with the Transparency Service, to Register Signed Statements.
+Authentication and authorization is implementation-specific, and out of scope of the SCITT Architecture.
+Signed Statements may be registered by a different Client identity than the Issuer.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:556]:
```suggestion
```

Comment (OR13) [inline draft-ietf-scitt-architecture.md:556]:
```suggestion
```

Comment (OR13) [reply draft-ietf-scitt-architecture.md:556]: less confusing to just say nothing here.

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (roywill) [reply draft-ietf-scitt-architecture.md:197]: I like this suggestion Orie.

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -544,12 +549,12 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
...
+1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity, which may be different than the Client identity.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:557]: This is registration policy details.... and interacts with all the other requirements we have for registration policies.... starting to feel like its better to say nothing here.

File: draft-ietf-scitt-architecture.md (hunk 3 above)
Comment (roywill) [reply draft-ietf-scitt-architecture.md:556]: Must a client authenticate?

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (ad-l) [reply draft-ietf-scitt-architecture.md:197]: the definition is OAuth specific ("protected resource" is an OAuth term) and needs to be adapted.

>An application making registration requests to a Transparency Service on behalf of a person holding a claim (who may or may not be the issuer).

Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:197]: Re my previous recommendation around RP was based in the applicability of OAUTH terms and concepts, so in that vein, I support this definition of client for the time being.

File: draft-ietf-scitt-architecture.md (hunk 1 above)
Comment (ad-l) [reply draft-ietf-scitt-architecture.md:198]: With my proposed definition above I would just add 

> (who may or may not be the Issuer of the statement)

at the end of my proposed definition above

File: draft-ietf-scitt-architecture.md (hunk 3 above)
Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:556]: Not that I have a definitive answer, but I can envision some uses where authentication is not desirable, but it really depends on what authentication means in this context.

File: draft-ietf-scitt-architecture.md (hunk 1 above)
Comment (aj-stein-nist) [reply draft-ietf-scitt-architecture.md:198]: I still have concerns given the explanations to answer and fully resolve 102, but my questions have been answered and they are outside the scope of my support of the PR recommendation and tentative changes proposed in https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/197/files#r1488069804.

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md]: I've updated above text

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -192,6 +192,12 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
+
+: an application making protected Transparency Service resource requests on behalf of the resource owner and with its authorization {{RFC6749}}.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:197]: Are you intending to make all SCITT clients OAUTH clients?

Seems like you just want to borrow the definition

Comment (OR13) [reply draft-ietf-scitt-architecture.md:197]: The current suggested text is likely to trigger a "should be a normative reference" comment.

Comment (ad-l) [reply draft-ietf-scitt-architecture.md:197]: agreed with Orie

File: draft-ietf-scitt-architecture.md [hunk 6]
```diff
@@ -192,6 +192,12 @@ Auditor:
...
+The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other
+      devices).
```
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:199]: Why make this confusing? Is that an oauth thing?

File: draft-ietf-scitt-architecture.md [hunk 7]
```diff
@@ -544,12 +550,11 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
...
 
-The same Signed Statement may be independently registered in multiple Transparency Services.
 To register a Signed Statement, the Transparency Service performs the following steps:
 
-1. **Client authentication:** This is implementation-specific and MAY be unrelated to the Issuer identity.
-Signed Statements may be registered by a different party than their Issuer.
-1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity.
+1. **Client authentication:** A Client authenticates with the Transparency Service, to Register Signed Statements.
+Authentication and authorization is implementation-specific, and out of scope of the SCITT Architecture.
+1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity, which may be different than the Client identity.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:557]:
```suggestion
```

**Review by ad-l — APPROVED** [review]

File: draft-ietf-scitt-architecture.md [hunk 8]
```diff
@@ -91,10 +91,10 @@ informative:
 
   RFC2397: DataURLs
   RFC6024:
+  RFC6749:
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:94]:
```suggestion
```

File: draft-ietf-scitt-architecture.md [hunk 9]
```diff
@@ -192,6 +191,12 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
+
+: an application making protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
+The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:197]:
```suggestion
The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
```

File: draft-ietf-scitt-architecture.md [hunk 10]
```diff
@@ -192,6 +191,12 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
+
+: an application making protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
+The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
+      devices).
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:198]:
```suggestion
```

File: draft-ietf-scitt-architecture.md [hunk 11]
```diff
@@ -192,6 +191,12 @@ Auditor:
 
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
+Client:
```
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:194]: I believe this role is outside the scope of SCITT and hence this PR should be abandoned!

**Review by OR13 — APPROVED** [review]

**Review by henkbirkholz — APPROVED** [review]
good enough
